### PR TITLE
merge overlap (initial)

### PR DESCRIPTION
**Pull request by teresi**
- add merge.py & tests for summing overlaps (initial)
- refactor functions into __init__.py
- add placeholders for future implementation (TODO SCOTT)

### CODE REVIEW

**Comment by teresi** [comment]
this is an initial implementation but I wanted to get it online since it's been a while
I refactored a few things into __init__
also, there are a few tags of `TODO SCOTT` that I'd like you to address

File: transposon/gene_data.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ def __init__(self, gene_dataframe, logger=None):
         self.stops = self.data_frame.Stop.to_numpy(copy=False)
         self.lengths = self.data_frame.Length.to_numpy(copy=False)
         self.chromosomes = self.data_frame.Chromosome.to_numpy(copy=False)
+        self.genome_id = "DEFAULT_GENOME_NAME"
+        # TODO SCOTT pls implement a variable for genome identifier, e.g. camarosa?
```
Comment (huckleberry-hound) [inline transposon/gene_data.py:43]: I will use the `g_fname` variable from the bottom of `__main__` in `density.py` for that. Sound good?

File: transposon/transposon_data.py [hunk 2]
```diff
@@ -42,6 +42,8 @@ def __init__(self, transposable_elements_dataframe, logger=None):
         self.orders = self.data_frame.Order.to_numpy(copy=False)
         self.superfamilies = self.data_frame.SuperFamily.to_numpy(copy=False)
         self.chromosomes = self.data_frame.Chromosome.to_numpy(copy=False)
+        self.genome_id = "DEFAULT_GENOME_NAME"
+        # TODO SCOTT pls implement a variable for genome identifier, e.g. camarosa?
```
Comment (huckleberry-hound) [inline transposon/transposon_data.py:46]: Same comment as above in `gene_data.py`

File: transposon/gene_data.py (hunk 1 above)
Comment (teresi) [reply transposon/gene_data.py:43]: yes, and in another branch pls

File: transposon/transposon_data.py (hunk 2 above)
Comment (teresi) [reply transposon/transposon_data.py:46]: yes, and in another branch pls

**Pull request merged**